src/index.tsx

```diff
@@ -84,90 +84,118 @@ const styles = StyleSheet.create({
-  onLeftSwipe?: Function,
-  onUpSwipe?: Function,
-  onSwipeCancelled?: Function,
-  hasUpAction: boolean,
-  leftText?: string,
-  leftView?: React.Element<*>,
-  loop: boolean,
-  upText: string,
-  upView?: React.Element<*>,
-  noView?: React.Element<*>,
-  onLoop: Function,
-  renderCard: Function,
-  renderLeftButton: Function,
-  renderNoMoreCards: Function,
-  renderRightButton: Function,
-  rightText?: string,
-  rightView?: React.Element<*>,
-  showUp: boolean,
-  showRight: boolean,
-  showleft: boolean,
-  smoothTransition: boolean,
-  stack: boolean,
-  stackDepth: number,
-  stackGuid?: string,
-  stackOffsetX: number,
-  stackOffsetY: number,
+export type SwipeDirection = 'left' | 'up' | 'right'
+export type SwipeHandler = (
+  card: any,
+  method: 'swipe' | 'button_press',
+) => boolean | Promise<boolean>
+export type ButtonRenderer = (
+  { onPress, disabled }: { onPress: () => void; disabled: boolean },
+) => React.ReactNode
+
+export interface Props {
+  allowGestureTermination?: boolean
+  cardKey: string
+  cards: Array<any>
+  dragY?: boolean
+  draggingDisabled?: boolean
+  guid?: number
+  hasUpAction?: boolean
+  leftStyle?: ViewStyle
+  leftText?: string
+  leftTextStyle?: TextStyle
+  leftView?: React.ReactNode
+  loop?: boolean
+  noView?: React.ReactNode
+  onCardRemoved?: (card: any, direction: SwipeDirection) => boolean
+  onDragRelease?: (direction?: SwipeDirection) => void
+  onDragStart?: () => void
+  onLeftPress?: () => void
+  onLeftSwipe?: SwipeHandler
+  onLoop?: () => void
+  onRightPress?: () => void
+  onRightSwipe?: SwipeHandler
+  onSwipeCancelled?: (card: any) => boolean
+  onUpSwipe?: SwipeHandler
+  renderCard: React.StatelessComponent
+  renderLeft?: (pan: Animated.ValueXY) => React.ReactNode
+  renderLeftButton: ButtonRenderer
+  renderNoMoreCards: () => React.ReactNode
+  renderRight?: (pan: Animated.ValueXY) => React.ReactNode
+  renderRightButton?: ButtonRenderer
+  renderUp?: (pan: Animated.ValueXY) => React.ReactNode
+  renderUpButton?: ButtonRenderer
+  rightStyle?: ViewStyle
+  rightText?: string
+  rightTextStyle?: TextStyle
+  rightView?: React.ReactNode
+  showRight?: boolean
+  showUp?: boolean
+  showleft?: boolean
+  smoothTransition?: boolean
+  stack?: boolean
+  stackDepth?: number
+  stackGuid?: string
+  stackOffsetX?: number
+  stackOffsetY?: number
+  upStyle?: ViewStyle
+  upText?: string
+  upTextStyle?: TextStyle
+  upView?: React.ReactNode
+}
+export type AnimatedValue = Animated.Value & { _value: number }
+export type AnimatedValueXY = Animated.ValueXY & {
+  x: AnimatedValue
+  y: AnimatedValue
 }
-type State = {
-  pan: Animated.valueXY,
-  enter: number,
-  cards: Array<Object>,
-  card: Object,
+export interface State {
+  pan: AnimatedValueXY
+  enter: Animated.Value
+  cards: Array<any>
+  card: any
 }
+
 export default class SwipeCards extends Component<Props, State> {
-  static defaultProps = {
+  public static defaultProps: Partial<Props> = {
```

Comment:
TypeScript doesn't handle React's defaultProps as well as Flow does; none of these are potentially undefined, even if we mark them as optional in the `Props` interface above, but TS doesn't realize that. In order to sidestep that issue, there's a little thing I'll point out further down